pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -507,11 +507,12 @@ func (s *DiscoveryServer) WorkloadUpdate(id string, labels map[string]string, _
 	}
 	w, f := s.WorkloadsByID[id]
 	if !f {
-		// First time this workload has been seen. Likely never connected, no need to
-		// push
+		// First time this workload has been seen. Maybe after the first connect, do a full push.
 		s.WorkloadsByID[id] = &Workload{
 			Labels: labels,
 		}
+		adsLog.Infof("First time sidecar instance added, full push %s ", id)
+		s.ConfigUpdate(true)
```

Comment:
do we have to push to Everyone? is there a way to push to just this sidecar? Otherwise, every time a new pod connects, everyone will have a full config push.

Reply:
Yes, I was thinking how to push only to this sidecar. Will try.